### PR TITLE
Science Module Complete

### PR DESCRIPTION
**Pull request by sharduld1908**
Hello,

Raising a pull request with the code added for the Science module #158

With this module added, the user will be able to use Science.h as a include file and generate data for 2 additional user-defined datatypes 'ChemicalElement' and 'Unit'
Just like the other modules, Science::chemicalElement() function will return a random element from the periodic table.
Also using function Science::unit() function will return a random fundamental unit for measuring distance, time, mass, temp, current.
I have also added functions that can generate units specifically for distance, time, mass, temp, current respectively.

Regards,
Shardul

### CODE REVIEW

**Comment by sharduld1908** [comment]
Not sure why a Unit Test from Vehicles module is failing. Trying to fix it. Any idea why this might be happening for clang compiler?

**Comment by cieslarmichal** [comment]
I ran tests again and it passed

File: include/faker-cxx/Science.h [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    {
+    }
+
+    bool operator==(const Unit& obj) const
+    {
+        return (this->name == obj.name && this->symbol == obj.symbol && this->usedToMeasure == obj.usedToMeasure);
+    }
+};
+
+class Science
```
Comment (cieslarmichal) [inline include/faker-cxx/Science.h:61]: Please add doxygen comments like any other module have :)

**Review by cieslarmichal — CHANGES_REQUESTED** [review]
Great job! Just add documentation to API amd we are good to merge

File: include/faker-cxx/Science.h (hunk 1 above)
Comment (sharduld1908) [reply include/faker-cxx/Science.h:61]: Alright will do

Comment (sharduld1908) [reply include/faker-cxx/Science.h:61]: Added comments. Please check.

**Comment by cieslarmichal** [comment]
Please add @code and @endcode lines in examples: check other api files please

**Review by cieslarmichal — APPROVED** [review]
Nice!

**Comment by sharduld1908** [comment]
Thanks for letting me contribute to this project. Learned a lot about open source contribution.

**Pull request merged**